### PR TITLE
tsdb: record a histogram of replay tsdb times

### PR DESCRIPTION
**Pull request by thorfour**
Signed-off-by: Thor <thansen@digitalocean.com>

**What this PR does**:
Adds a histogram to record tsdb replay timings.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/metrics.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type ingesterMetrics struct {
 	queriedSamples        prometheus.Histogram
 	queriedSeries         prometheus.Histogram
 	queriedChunks         prometheus.Histogram
+	tsdbReplayTime        prometheus.Histogram
```
Comment (pstibrany) [inline pkg/ingester/metrics.go]: I think this should go to `TSDBState` struct in ingester_v2.go. Also technically, it's "open" time for existing DBs, so perhaps "tsdbStartupOpenTime" (?)

Comment (thorfour) [reply pkg/ingester/metrics.go]: Yea that's fair, I forgot we had metrics in that struct.

Comment (thorfour) [reply pkg/ingester/metrics.go]: Well, actually, should we have metrics in there? Or should we move all metrics into the metrics struct?

Comment (pstibrany) [reply pkg/ingester/metrics.go]: I think blocks-specific metrics should go there. Chunks and shared metrics between chunks and blocks go to metrics.go

Comment (thorfour) [reply pkg/ingester/metrics.go]: I get that delineation, but from another perspective it feels strange to have a metrics file that doesn't contain all the metrics.

Comment (pracucci) [reply pkg/ingester/metrics.go]: My feedback on the topic:

1. `tsdbMetrics struct` is a different thing and it's used for the Thanos mapped metrics. I believe this metric should go to `TSDBState struct` where we already have other blocks-storage related metrics tracked in Cortex (ie. `compactionsTriggered`).
2. The `tsdb` prefix in the `tsdbReplayTime` is redundant and should be removed. We could call it `walReplayDuration` which is that it is (I understand Peter's feedback that technically it's TSDB open, but practically the time there is taken by WAL replay and we track this metric only for existing TSDBs and not new ones we just open)

Then, we can discuss whether it's a good or bad things having ingester metrics defined in multiple places, but this shouldn't be address by this PR (we can discuss it offline and eventually come up with a dedicated PR to address it).

File: pkg/ingester/metrics.go [hunk 2]
```diff
@@ -184,6 +185,11 @@ func newIngesterMetrics(r prometheus.Registerer, createMetricsConflictingWithTSD
 			Name: "cortex_oldest_unflushed_chunk_timestamp_seconds",
 			Help: "Unix timestamp of the oldest unflushed chunk in the memory",
 		}),
+		tsdbReplayTime: promauto.With(r).NewHistogram(prometheus.HistogramOpts{
+			Name:    "cortex_ingester_tsdb_replay_time_seconds",
```
Comment (pracucci) [inline pkg/ingester/metrics.go]: `cortex_ingester_tsdb_replay_time_seconds` > `cortex_ingester_tsdb_wal_replay_duration_seconds`

File: pkg/ingester/metrics.go [hunk 3]
```diff
@@ -184,6 +185,11 @@ func newIngesterMetrics(r prometheus.Registerer, createMetricsConflictingWithTSD
...
+			Help:    "The total time it takes to open and replay a TSDB.",
```
Comment (pracucci) [inline pkg/ingester/metrics.go]: `The total time it takes to open and replay a TSDB WAL.`

File: pkg/ingester/metrics.go (hunk 1 above)
Comment (pracucci) [reply pkg/ingester/metrics.go]: Other thoughts. We already have `cortex_ingester_wal_replay_duration_seconds` metric (chunks storage). Why don't we reuse the same?

Comment (thorfour) [reply pkg/ingester/metrics.go]: Only problem with that is that's a gauge and not a histogram.

**Pull request merged**